core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -891,10 +891,12 @@ private[spark] object SparkSubmitUtils {
     val localIvy = new FileSystemResolver
     val localIvyRoot = new File(ivySettings.getDefaultIvyUserDir, "local")
     localIvy.setLocal(true)
-    localIvy.setRepository(new FileRepository(localIvyRoot))
```

Comment:
It's redundant because the filesystem patterns defined below use an absolute path (with the `localIvyRoot`) and the resolver then ignores the repository root set here.  Only if it determines the patterns are using relative paths it will then prefix them with the root.   It doesn't hurt anything to leave it in though, if you prefer.